Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs

```diff
@@ -121,7 +123,7 @@ findAllDriversWithInfoAndVehicle merchantId limitVal offsetVal mbVerified mbEnab
       p <- catMaybes <$> mapM fromTType' persons
       di <- catMaybes <$> mapM fromTType' driverInfos
       v <- mapM (maybe (pure Nothing) fromTType') vehicles
-      pure $ zip3 p di v
+      pure $ filter (\(person, _, _) -> person.merchantOperatingCityId == moCityId) $ zip3 p di v
```

Comment:
Why don't you add inside the query only, instead of filtering later on.

Reply:
Done